### DISCUSSION TITLE
How to call wasm API from native library

### DISCUSSION THREAD

**Discussion by pmaskale** [bytecodealliance/wasm-micro-runtime#1698] [Q&A]
Can we call the API defined in wasm-app from native library.
basically by running iwasm --native-lib=mylib.so  mywasm.wasm 

by loading wasm module to buffer and creating exec_env environment  and lookup for  the function is the only way to call it or is there any other way?

**Comment by wenyongh** [comment]
Hi, you can also try wasm-c-api, refer to the document:
  https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/doc/wasm_c_api.md
And the samples:
  https://github.com/WebAssembly/wasm-c-api/tree/main/example
